src/components/spectrogram/spectrogram.ts

```diff
@@ -566,33 +591,59 @@ export class SpectrogramComponent extends SignalWatcher(ChromeHost(LitElement))
       // if the user starts playing the audio, stops playing it, then starts playing it again within the same frame
       // we would have two animation requests, and both would continue polling the time
       // in all subsequent frames, we would have two time updates per frame
-      if (this.nextRequestId) {
-        window.cancelAnimationFrame(this.nextRequestId);
-        this.nextRequestId = null;
+      if (this.timeInterpolateFrameId !== null) {
+        window.cancelAnimationFrame(this.timeInterpolateFrameId);
+        this.timeInterpolateFrameId = null;
       }
 
-      const initialTime = this.highAccuracyElapsedTime() - this.currentTime.peek();
-      this.nextRequestId = requestAnimationFrame(() => this.pollUpdateHighAccuracyTime(initialTime));
+      // we use peek() here because we do not want to create a subscription
+      // to the currentTime signal
+      const initialTime = this.highAccuracyElapsedTime() - this._currentTime.peek();
+      this.timeInterpolateFrameId = requestAnimationFrame(() => this.pollUpdateHighAccuracyTime(initialTime));
+
       return;
     }
 
     // even though the audio elements currentTime is not accurate to perform
     // animations, we can use the currentTime to if we only want to update the
     // audio once without polling updates
-    this.currentTime.value = this.mediaElement.currentTime;
+    this._currentTime.value = this.mediaElement.currentTime;
   }
 
-  // TODO: move somewhere else
-  private nextRequestId: number | null = null;
-
   private pollUpdateHighAccuracyTime(startTime: number): void {
     if (!this.paused) {
```

Comment:
consider setting _currentTime = this.mediaElement.currentTime when not playing (the else case of this branch)

Reply:
![image](https://github.com/user-attachments/assets/17609bfa-ddd4-40cf-9388-1f3e8950be17)